scenarios/kubernetes_e2e.py

```diff
@@ -0,0 +1,239 @@
+
+    # exec
+    container = '%s-%s' % (os.environ.get('JOB_NAME'), os.environ.get('BUILD_NUMBER'))
+
+    print 'Starting...'
+
+    cmd = ['timeout', '-s', 'KILL', docker_timeout, 'docker', 'run', '--rm',
+    # For local test, timeout not work on mac
+    # cmd = ['docker', 'run', '--rm',
+      '--name=%s' % container,
+      '-v', '%s/_artifacts":/workspace/_artifacts' % workspace,
+      '-v', '/etc/localtime:/etc/localtime:ro'
+    ]
+
+    if os.environ.get('JENKINS_GCE_SSH_PRIVATE_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/google_compute_engine:ro'
+            % os.environ['JENKINS_GCE_SSH_PRIVATE_KEY_FILE']])
+
+    if os.environ.get('JENKINS_GCE_SSH_PUBLIC_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/google_compute_engine.pub:ro'
+            % os.environ['JENKINS_GCE_SSH_PUBLIC_KEY_FILE']])
+
+    if os.environ.get('JENKINS_AWS_SSH_PRIVATE_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/kube_aws_rsa:ro'
+            % os.environ['JENKINS_AWS_SSH_PRIVATE_KEY_FILE']])
+
+    if os.environ.get('JENKINS_AWS_SSH_PUBLIC_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/kube_aws_rsa.pub:ro'
+            % os.environ['JENKINS_AWS_SSH_PUBLIC_KEY_FILE']])
+
+    if os.environ.get('JENKINS_AWS_CREDENTIALS_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.aws/credentials:ro'
+            % os.environ['JENKINS_AWS_CREDENTIALS_FILE']])
+
+    if os.environ.get('GOOGLE_APPLICATION_CREDENTIALS'):
+        cmd.extend(['-v', '%s:/service-account.json:ro'
+            % os.environ['GOOGLE_APPLICATION_CREDENTIALS'],
+                    '-e', 'GOOGLE_APPLICATION_CREDENTIALS=/service-account.json'])
+
+    for key, value in os.environ.items():
+        if key not in docker_env_ignore:
+            cmd.extend(['-e', '%s=%s' % (key, value)])
+
+    cmd.extend(docker_extra_args)
+
+    cmd.extend([
+      '-e', 'HOME=/workspace',
+      '-e', 'WORKSPACE=/workspace',
+      'gcr.io/k8s-testimages/kubekins-e2e:%s' % e2e_image_tag
+    ])
+
+    try:
+        check(*cmd)
+        print 'Finished Successfully'
+    except subprocess.CalledProcessError as exc:
+        print 'Exiting with code: %d' % exc.returncode
+        if exc.returncode == 137:
+            # docker runs containers with session id being the same as the pid
+
+            # TODO(@krzyzacy) how to handle this..
+            # pid = $(docker inspect --format '{{.State.Pid}}' "${container}")
+            # print 'Processes running under container %s' % container
+            # check('ps', 'wwuf', '--forest', '--sid', "${pid}"
+            try:
+                check('docker', 'stop', container)
+            except subprocess.CalledProcessError:
+                pass
+
+            try:
+                check('sudo', 'chmod', 'a+rX', '-R' '%s/_artifacts/' % workspace)
+            except subprocess.CalledProcessError:
+                pass
+
+
+if __name__ == '__main__':
+    # Arbitrary args
```

Comment:
I was intended to make all upper cased arg as automated envs so I can avoid creating a `ci-kubernetes-foo.env` file.